teuthology/task/install/rpm.py

```diff
@@ -143,6 +145,16 @@ def _downgrade_packages(ctx, remote, pkgs, pkg_version, config):
     remote.run(args='sudo yum -y downgrade {}'.format(' '.join(pkgs_opt)))
     return [pkg for pkg in pkgs if pkg not in downgrade_pkgs]
 
+def _retry_if_503_in_output(remote, args):
+    # wait at most 5 minutes
+    with safe_while(sleep=10, tries=30) as proceed:
+        while proceed():
+            stdout = StringIO()
+            try:
+                remote.run(args=args, stdout=stdout)
```

Comment:
(wonderring) maybe there is make sense to implement a _retry_ option for run method, or at least remote.

Reply:
If we implement the retry code in `run` method, I think we need to pass `retry_cnt`, `wait_seconds` and `func()` for defferent use cases. The `func()` will tell the `run` method in which case it need to retry.